### PR TITLE
Refactor CopilotModal + UI tests

### PR DESCRIPTION
**Pull request by mohebifar**
🌟**Goal/Reason/Feature for this PR:** 🌟
This PR contains a refactor of the main parts of the UI. It allows you to choose between the two ways of rendering the overlay: `SVG`; and the `View`. With the SVG overlay, you can have a nice and smooth animation for the transitions between the steps. If you don't want to add a native package that is react-native-svg, you can stick with the `View` overlay.

Imgur

```js
copilot({
  overlay?: 'view' | 'svg',
  animated?: bool,
})
```

Also, the tooltip component can be customized now.
```js
copilot({
  tooltipComponent?: ({
    isFirstStep,
    isLastStep,
    handleNext,
    handlePrev,
    handleStop,
    currentStep,
  }) => {
      // return ...
  }
})(WrappedComponent)
```
It also adds UI tests.

💃**Changes Made:** 💃 
- Separated the overlay from the modal component ✂️
- Added an optional dependency to react-native-svg 🌁
- Added a new overlay component using SVG 🎨
- Made the overlay animatable 📽
- The tooltip/balloon component can now be customized 💭
- Added UI tests using react-test-renderer ✅

 🏗**TODO:** 🏗
 
- [x] Add tests for the copilot package
- [x] Bump the version to `2.0.0`
- [x] Update docs

### CODE REVIEW

**Comment by RichardLitt** [comment]
@mohebifar Can you describe what this PR is doing a bit more, so that any community members and watchers who are following along may be able to weigh in with reviews, too?

**Comment by mohebifar** [comment]
Updated the description

**Comment by RichardLitt** [comment]
@mohebifar Thanks! That helps a lot. :)

File: src/components/CopilotModal.js [hunk 1]
```diff
@@ -114,111 +115,98 @@ class CopilotModal extends Component<Props, State> {
...
   render() {
+    const { tooltipComponent: TooltipComponent } = this.props;
+
+    /* eslint-disable global-require */
+    const MaskComponent = this.props.overlay === 'svg'
+      ? require('./SvgMask').default
+      : require('./ViewMask').default;
+    /* eslint-enable */
+
     return this.props.visible ? (
```
Comment (cfnelson) [inline src/components/CopilotModal.js]: Is there a reason why you wouldn't want this check before `MaskComponent` to return null without loading the svg or view mask? e.g when `this.props.visble === false`?  assumption that `visible` is a bool e.g `isVisible`. 

Example of what I mean:
```
if (!this.props.visible) {
  return null;
}

const MaskComponent = ...

return (
  <View ... > ...</View>
);
```

File: example/package.json [hunk 2]
```diff
@@ -3,24 +3,31 @@
...
   },
   "jest": {
     "preset": "jest-expo"
   },
   "dependencies": {
-    "expo": "^21.0.0",
-    "react": "16.0.0-alpha.12",
-    "react-native": "^0.48.4"
+    "expo": "^26.0.0",
+    "react": "16.3.0-alpha.1",
```
Comment (cfnelson) [inline example/package.json]: React `16.3.1` was released about 10 days ago. Wondering if there is a reason why you have fixed React to an alpha version?

File: src/components/SvgMask.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      canvasSize: { x: 0, y: 0 },
+      size: new Animated.ValueXY({ x: 0, y: 0 }),
+      position: new Animated.ValueXY({ x: 0, y: 0 }),
+      animated: false,
+    };
+
+    this.state.position.addListener(this.animationListener);
+  }
+
+  componentWillReceiveProps(nextProps) {
```
Comment (cfnelson) [inline src/components/SvgMask.js]: Noticed you were using React `0.16.3`. Wondering if you should/can swap the deprecated `componentWillReceiveProps` out with the new lifecycle -> `getDerivedStateFromProps` which was made to replace this.
Check out the details from React [here](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html)

File: src/components/ViewMask.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  canvasSize: valueXY,
+};
+
+class ViewMask extends Component<Props, State> {
+  state = {
+    size: new Animated.ValueXY({ x: 0, y: 0 }),
+    position: new Animated.ValueXY({ x: 0, y: 0 }),
+  };
+
+  componentWillReceiveProps(nextProps) {
```
Comment (cfnelson) [inline src/components/ViewMask.js]: Same as above, maybe think of replacing with `getDerivedStateFromProps`?

Check out the details from React [here](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html)

File: example/scripts/watch-and-copy-src.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+// Copied from https://github.com/leecade/react-native-swiper/blob/3f688747264a09f8fd9893037912d4c838cceeb2/examples/scripts/watch-and-copy-src.js
```
[the remark this answers is not in the document]
Comment (mohebifar) [reply example/scripts/watch-and-copy-src.js]: There are some small changes. I honestly don't know. What do you think?

**Pull request merged**